app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/prometheus_client_mixin.rb

```
@@ -59,17 +50,11 @@ def labels_to_s(labels, job = "kubernetes-cadvisor")
 
   def prometheus_try_connect
     begin
-      response = prometheus_client.get("query", :query => "ALL")
+      data = prometheus_client.query(:query => "ALL")
     rescue StandardError => err
       raise MiqException::MiqUnreachableError, err.message, err.backtrace
     end
 
-    begin
-      data = JSON.parse(response.body)
-    rescue StandardError => err # if auth_proxy fail it returns an html doc
-      raise MiqException::MiqInvalidCredentialsError, err.message, err.backtrace
```

Where is the credential error handled now? If it is in the client then line 55 will turn it intu `MiqUnreachableError`

---

👍 nice catch, now that we have a specialized client we can leverage it's error's to get better differentiations between specific types of errors.

Thanks @enoodle , will remove the code that hide the real error.